sdk/identity/azure-identity/azure/identity/_internal/shared_token_cache.py

```diff
@@ -88,13 +89,19 @@ class SharedTokenCacheBase(ABC):
     def __init__(self, username=None, **kwargs):  # pylint:disable=unused-argument
         # type: (Optional[str], **Any) -> None
 
-        authority = kwargs.pop("authority", None)
-        self._authority = normalize_authority(authority) if authority else get_default_authority()
-
-        environment = urlparse(self._authority).netloc
-        self._environment_aliases = KNOWN_ALIASES.get(environment) or frozenset((environment,))
-        self._username = username
-        self._tenant_id = kwargs.pop("tenant_id", None)
+        self._auth_record = kwargs.pop("authentication_record", None)  # type: Optional[AuthenticationRecord]
+        if self._auth_record:
+            self._authority = self._auth_record.authority
```

Comment:
I don't see we have normalize_authority(authority) in ctor of AuthenticationRecord. So maybe we need it here

Reply:
Accounts in MSAL's cache don't have "normalized" authorities, so neither does `AuthenticationRecord`. This credential's client will normalize the authority it's given.